app/connector/kafka/src/main/java/io/syndesis/connector/kafka/service/KafkaBrokerServiceImpl.java

```diff
@@ -83,22 +87,66 @@ public void ping() throws KafkaBrokerServiceException {
         }
     }
 
-    private static Properties getKafkaAdminClientConfiguration(String brokers, String transportProtocol, KeyStoreHelper brokerKeyStoreHelper) {
+    private Properties getKafkaAdminClientConfiguration(String brokers, String securityProtocol, KeyStoreHelper brokerKeyStoreHelper) {
         Properties properties = new Properties();
-        properties.put("bootstrap.servers", brokers);
-        properties.put("connections.max.idle.ms", 10000);
-        properties.put("request.timeout.ms", 5000);
-        if (!"PLAINTEXT".equals(transportProtocol)) {
-            properties.put("security.protocol", "SSL");
-            if (brokerKeyStoreHelper != null) {
-                properties.put("ssl.endpoint.identification.algorithm", "");
-                properties.put("ssl.keystore.location", brokerKeyStoreHelper.getKeyStorePath());
-                properties.put("ssl.keystore.password", brokerKeyStoreHelper.getPassword());
-                properties.put("ssl.key.password", brokerKeyStoreHelper.getPassword());
-                properties.put("ssl.truststore.location", brokerKeyStoreHelper.getKeyStorePath());
-                properties.put("ssl.truststore.password", brokerKeyStoreHelper.getPassword());
+        properties.put(CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG, brokers);
+        properties.put(CommonClientConfigs.CONNECTIONS_MAX_IDLE_MS_CONFIG, 10000);
+        properties.put(CommonClientConfigs.REQUEST_TIMEOUT_MS_CONFIG, 5000);
+        properties.put(CommonClientConfigs.SECURITY_PROTOCOL_CONFIG, securityProtocol);
+        properties.put(CommonClientConfigs.DEFAULT_API_TIMEOUT_MS_CONFIG, 5000);
+        // use certificate to authenticate to kafka server
+        if (brokerKeyStoreHelper != null) {
+            properties.put(SslConfigs.SSL_ENDPOINT_IDENTIFICATION_ALGORITHM_CONFIG, "");
+            properties.put(SslConfigs.SSL_KEYSTORE_LOCATION_CONFIG, brokerKeyStoreHelper.getKeyStorePath());
+            properties.put(SslConfigs.SSL_KEYSTORE_PASSWORD_CONFIG, brokerKeyStoreHelper.getPassword());
+            properties.put(SslConfigs.SSL_KEY_PASSWORD_CONFIG, brokerKeyStoreHelper.getPassword());
+            properties.put(SslConfigs.SSL_TRUSTSTORE_LOCATION_CONFIG, brokerKeyStoreHelper.getKeyStorePath());
+            properties.put(SslConfigs.SSL_TRUSTSTORE_PASSWORD_CONFIG, brokerKeyStoreHelper.getPassword());
+        } else if (StringUtils.isNotBlank(username) && StringUtils.isNotBlank(password)) {
```

Comment:
Let's not add a dependency we might not need

```suggestion
        } else if (ObjectHelper.isNotEmpty(username) && ObjectHelper.isNotEmpty(password)) {
```

Reply:
Thanks for the tip, I will change it.